x-pack/platform/plugins/shared/lens/public/editor_frame_service/editor_frame/config_panel/esql_editor.tsx

```diff
@@ -224,7 +226,7 @@ export function ESQLEditor({
         <ESQLDataGridAccordion
           dataGridAttrs={dataGridAttrs}
           isAccordionOpen={isESQLResultsAccordionOpen}
-          isTableView={attributes?.visualizationType !== 'lnsDatatable'}
+          isTableView={visualization.activeId !== 'lnsDatatable'}
```

Comment:
A lot of this ES|QL editor code uses `attributes`. I was hoping to get rid of this to pull from the state but decided not to for ES|QL.

In this case the `visualizationType` will be out of sync when the chart type is switched. This still does not fix deeper issues in `ESQLDataGridAccordion` that uses `isTableView` to set the `activeColumns` but it's a start.